lib/ansible/module_utils/facts/system/distribution.py

```diff
@@ -228,10 +228,11 @@ def parse_distribution_file_Slackware(self, name, data, path, collected_facts):
     def parse_distribution_file_Amazon(self, name, data, path, collected_facts):
         amazon_facts = {}
         if 'Amazon' not in data:
-            # return False  # TODO: remove   # huh?
-            return False, amazon_facts  # TODO: remove
+            return False, amazon_facts
         amazon_facts['distribution'] = 'Amazon'
-        amazon_facts['distribution_version'] = data.split()[-1]
+        version = re.search(r" ([\d\.]+) ?", data)
```

Comment:
Rather than use a regexp, split the string and get the digit: 
```suggestion
        version = [n for n in data.split() if n.isdigit()]
```